cli/remote_build.go

```diff
@@ -729,76 +721,79 @@ func (cmd *RemoteCmd) submitBuild(project *types.Project, tagMap map[string]stri
 	case 500:
 		return fmt.Errorf("Internal server error")
 	}
-
-	response := string(body)
-
 	//Process simple response from the API
-	response = strings.ReplaceAll(response, "\"", "")
-
-	url := ""
-	if strings.Count(response, "\n") > 0 {
-		url = strings.Split(response, "\n")[0]
-	} else {
-		url = response
+	body = bytes.ReplaceAll(body, []byte(`"`), nil)
+	if i := bytes.IndexByte(body, '\n'); i != -1 {
+		body = body[:i]
+	}
+	logURL, err := url.Parse(string(body))
+	if err != nil {
+		return fmt.Errorf("server response: parse log URL: %w", err)
+	}
+	if logURL.Scheme != "ws" && logURL.Scheme != "wss" {
+		return fmt.Errorf("server response: parse log URL: unhandled scheme %q", logURL.Scheme)
+	}
+	uiURL := ""
+	if id, err := buildIDFromLogURL(logURL); err == nil {
+		// Fine to ignore error: this is displayed as a fallback if other things fail.
+		uiURL, _ = ybconfig.ManagementUrl("/" + project.OrgSlug + "/" + project.Label + "/builds/" + id)
```

Comment:
I agree it's fine to continue if `err != nil`, but we could get some value out of outputting it anyway, so we don't overlook something silently failing.

Reply:
Done.